### PR TITLE
fix: remove workloadCount from trial responses; single-trial view fix

### PR DESCRIPTION
**Pull request by mapmeld**
## Description

`workloadCount` was adding complexity to our queries with limited utility.  The PR removes `workloadCount` and instead makes a request for one trial workload (paginated workloads API).

If the trial finishes with 0 workloads then `trialNeverData` gets set to `true` and the TensorBoard button is hidden.

## Test Plan

On a multi-trial experiment, view a single trial. It should have a TensorBoard button in the top right.

## Checklist
- [x] Changes have been manually QA'd
- [x] User-facing API changes need the "User-facing API Change" label.
- [x] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [x] Licenses should be included for new code which was copied and/or modified from any external code.
- [x] If modifying `/webui/react/src/shared/` verify `make -C webui/react test-shared` passes.

### CODE REVIEW

File: master/test/integration/api/api_trials_test.go [hunk 1]
```diff
@@ -124,9 +124,8 @@ func trialDetailAPITests(
 			ctx, _ := context.WithTimeout(creds, 10*time.Second)
 			req := apiv1.GetTrialRequest{TrialId: int32(trial.ID)}
 
-			tlCl, err := cl.GetTrial(ctx, &req)
+			_, err = cl.GetTrial(ctx, &req)
 			assert.NilError(t, err, "failed to fetch api details")
-			assert.Equal(t, tlCl.Trial.WorkloadCount, int32(1), "mismatching workload length")
```
Comment (hamidzr) [inline master/test/integration/api/api_trials_test.go:129]: missing out on this test

File: webui/react/src/pages/TrialDetails/TrialDetailsHeader.tsx [hunk 2]
```diff
@@ -53,10 +50,23 @@ const TrialDetailsHeader: React.FC<Props> = ({
...
+      });
+      setTrialNeverData(wl.workloads.length === 0);
+    }
+  }, [ trial ]);
+
   const headerOptions = useMemo<Option[]>(() => {
     const options: Option[] = [];
 
-    if (!trialWillNeverHaveData(trial)) {
+    if (!trialNeverData) {
```
Comment (hamidzr) [inline webui/react/src/pages/TrialDetails/TrialDetailsHeader.tsx:69]: nit keep the same name? will avoid adding extra changes as well

File: webui/react/src/pages/TrialDetails/TrialDetailsHeader.tsx [hunk 3]
```diff
@@ -32,6 +28,7 @@ const TrialDetailsHeader: React.FC<Props> = ({
   trial,
 }: Props) => {
   const [ isRunningTensorBoard, setIsRunningTensorBoard ] = useState<boolean>(false);
+  const [ trialNeverData, setTrialNeverData ] = useState<boolean>(false);
```
Comment (hamidzr) [inline webui/react/src/pages/TrialDetails/TrialDetailsHeader.tsx:31]: do we need this? could the usememo just return the value you want? const trialNeverData = useMemo ...

**Review by ioga — APPROVED** [review]

File: webui/react/src/pages/TrialDetails/TrialDetailsHeader.tsx (hunk 3 above)
Comment (hamidzr) [reply webui/react/src/pages/TrialDetails/TrialDetailsHeader.tsx:31]: ah it's async so we do. maybe we then don't want it to be useMemo? that could be a useEffect I think

**Review by hamidzr — COMMENTED** [review]
I reviewed all the changes but the sql one. mostly looks good.  I haven't visually tested it yet

File: webui/react/src/pages/TrialDetails/TrialDetailsHeader.tsx (hunk 3 above)
Comment (mapmeld) [reply webui/react/src/pages/TrialDetails/TrialDetailsHeader.tsx:31]: I wasn't sure the right way to get the async value, and have it be usable inside `headerOptions`

File: master/test/integration/api/api_trials_test.go (hunk 1 above)
Comment (mapmeld) [reply master/test/integration/api/api_trials_test.go:129]: the original test prior was looking for `len(tlCl.Trial.Workloads) == 1`, which was replaced by `tlCl.Trial.WorkloadCount == 1`, but now has no equivalent

**Pull request merged**